### PR TITLE
Add ionbot reader; read ion mobility from mzid files

### PR DESCRIPTION
**Pull request by ArthurDeclercq**
### Added
- `io`: Add reader for ionbot csv PSM files
- `io.mzid`: Read `ion_mobility`

### Changed
- `io.mzid`: Slightly improved `rt_key` parsing functionality

### CODE REVIEW

File: psm_utils/io/ionbot.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            peptidoform=self._parse_peptidoform(
+                psm_dict["database_peptide"],
+                psm_dict["modifications"],
+                psm_dict["charge"],
+            ),
+            spectrum_id=psm_dict["spectrum_title"],
+            run=psm_dict["spectrum_file"],
+            is_decoy=psm_dict["database"] == "D",
+            score=float(psm_dict["psm_score"]),
+            # precursor_mz=float(psm_dict["m/z"]),
```
Comment (RalfG) [inline psm_utils/io/ionbot.py:76]: **todo:** Remove commented line.

File: psm_utils/io/ionbot.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            retention_time=float(psm_dict["observed_retention_time"]),
+            protein_list=psm_dict["proteins"].split(
+                "||"
+            ),  # what is the ionbot separator?
```
Comment (RalfG) [inline psm_utils/io/ionbot.py:80]: **question:** Is this resolved?

File: psm_utils/io/ionbot.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            source="Ionbot",
```
Comment (RalfG) [inline psm_utils/io/ionbot.py:81]: **nitpick:** "ionbot" is without capital `I`.

File: psm_utils/io/ionbot.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            qvalue=float(psm_dict["q-value"]),
+            pep=float(psm_dict["PEP"]),
+            provenance_data=({"Ionbot_filename": str(self.filename)}),
```
Comment (RalfG) [inline psm_utils/io/ionbot.py:84]: **nitpick:** "ionbot" is without capital `I`.

File: psm_utils/io/ionbot.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        pattern = re.compile(r"^(?P<U>\[\S*?\])?(?P<mod>.*?)(?P<AA>\[\S*?\])?$")
+
+        for position, label in zip(
+            modifications.split("|")[::2], modifications.split("|")[1::2]
+        ):
+            mod_match = pattern.search(label)
+
+            if mod_match.group("U"):
+                parsed_label = "U:" + mod_match.group("U")[1:-1]
+
```
Comment (RalfG) [inline psm_utils/io/ionbot.py:104]: **nitpick:** Remove blank line

File: psm_utils/io/ionbot.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        if charge:
+            proforma_seq += f"/{charge}"
+
+        return proforma_seq
+
+
+class InvalidPeprecError(PSMUtilsIOException):
+    """Invalid Peptide Record file."""
+
+    pass
```
Comment (RalfG) [inline psm_utils/io/ionbot.py:129]: **question:** Why is this defined here?

File: psm_utils/io/ionbot.py [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+
+class InvalidIonbotModificationError(InvalidPeprecError):
```
Comment (RalfG) [inline psm_utils/io/ionbot.py:132]: **question:** Why does this inherit from InvalidPeprecError and not from PSMUtilsIOException?

**Review by RalfG — CHANGES_REQUESTED** [review]

File: psm_utils/io/mzid.py [hunk 8]
```diff
@@ -250,7 +245,7 @@ def _get_peptide_spectrum_match(
             precursor_mz = None
 
         # Override spectrum-level RT if present at PSM level
-        if self._rt_key:
+        if self._rt_key in sii.keys():
```
[the remark this answers is not in the document]
Comment (ArthurDeclercq) [reply psm_utils/io/mzid.py:248]: No was a leftover from the other implementation with one rt_key for both spectrum and psm level

**Pull request merged**